esp/smc/SMCLib/TpWrapper.cpp

```diff
@@ -2410,29 +2421,36 @@ static void refreshValidTargets()
     }
 }
 
+static void configUpdate(const IPropertyTree *oldComponentConfiguration, const IPropertyTree *oldGlobalConfiguration)
+{
+    CriticalBlock block(validTargetSect);
+    // as much as effort [small] to check if different as to refresh
+    refreshValidTargets();
+    PROGLOG("Valid targets updated");
+}
+
 extern TPWRAPPER_API void validateTargetName(const char* target)
 {
     if (isEmptyString(target))
         throw makeStringException(ECLWATCH_INVALID_CLUSTER_NAME, "Empty target name.");
 
     CriticalBlock block(validTargetSect);
-    if (targetsDirty)
+#ifdef _CONTAINERIZED
+    if ((unsigned) -1 == reloadConifgCBId) // once only
     {
         refreshValidTargets();
-        targetsDirty = false;
+        reloadConifgCBId = installConfigUpdateHook(configUpdate);
```

Comment:
Should installConfigUpdateHook have a flag to indicate the function should be called when it is registered?
This could then be
reloadConfigHook.installOnce(configUpdate, true);
Could be an option purely on the the (new) hook class, rather than the install function.

Reply:
have implemented in https://github.com/hpcc-systems/HPCC-Platform/pull/14966